### PR TITLE
First commit to feature branch

### PR DESCRIPTION
**Pull request by tatsianabeliai**
Pull request for review

### CODE REVIEW

File: src/test/java/ClassOneTest.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    @BeforeClass
+    public static void onceExecutedBeforeAll() {
+        LOG.info("@BeforeClass: executedBeforeAllTests");
+    }
+
+    @Before
+    public void executedBeforeEachTest() {
+        LOG.info("@Before: executedBeforeEachTest");
+        Mathematics math = new Mathematics();
+        mathematics = math;
```
Comment (partyzaner) [inline src/test/java/ClassOneTest.java]: unnecessary math variable

File: src/test/java/ClassOneTest.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        try {
+            mathematics.setResult(1);
+            Thread.sleep(1500);
+            mathematics.deduct(numberOne, numberTwo);
+            int result = mathematics.getResult();
+            assertEquals("Not expected result is displayed", 45, result);
+        } catch (InterruptedException ex) {
+            LOG.error(ex);
+            Assert.fail();
+        }
```
Comment (partyzaner) [inline src/test/java/ClassOneTest.java]: 1) use try-catch only for lines that can throw exception
2) Assert.fail without message

File: src/test/java/ClassOneTest.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    }
+
+    @Category(SlowTests.class)
+    @Test
+    public void multiplyTest() {
+        mathematics.multiply(numberOne, numberTwo);
+        int result = mathematics.getResult();
+        assertEquals("Not expected result is displayed", 250, result);
```
Comment (partyzaner) [inline src/test/java/ClassOneTest.java]: magic number. add variable for 250

File: src/test/java/ClassOneTest.java [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    }
+
+    @Category(SlowTests.class)
+    @Test
+    public void divideTest() {
+        mathematics.divide(numberOne, numberTwo);
+        int result = mathematics.getResult();
+        assertEquals("Not expected result is displayed", 10, result);
```
Comment (partyzaner) [inline src/test/java/ClassOneTest.java]: magic number

Comment (partyzaner) [inline src/test/java/ClassOneTest.java]: magic number

File: src/test/java/ClassOneTest.java [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+    }
+
+    @Category(SlowTests.class)
+    @Test
+    public void divideByZeroTest() {
+        mathematics.divide(1, 0);
+        int result = mathematics.getResult();
+        assertEquals("Not expected result is displayed", 10, result);
```
Comment (partyzaner) [inline src/test/java/ClassOneTest.java]: magic number

File: src/test/java/DDTTest.java [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+
+import java.io.IOException;
+import java.util.Collection;
+
+import static junit.framework.TestCase.assertEquals;
+
+@RunWith(value = Parameterized.class)
+public class DDTTest {
+    private static final Logger LOG = Logger.getLogger(DDTTest.class);
+    static Mathematics mathematics;
```
Comment (partyzaner) [inline src/test/java/DDTTest.java]: Why static?

File: src/test/java/DDTTest.java [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+    @BeforeClass
+    public static void onceExecutedBeforeAll() {
+        LOG.info("@BeforeClass: executedBeforeAllTests");
+    }
+
+    @Before
+    public void executedBeforeEachTest() {
+        LOG.info("@Before: executedBeforeEachTest");
+        Mathematics math = new Mathematics();
+        mathematics = math;
```
Comment (partyzaner) [inline src/test/java/DDTTest.java]: Unnecessary math variable

File: src/test/java/ClassOneTest.java [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+import org.junit.runners.MethodSorters;
+
+import static org.junit.Assert.assertEquals;
+
+@FixMethodOrder(MethodSorters.NAME_ASCENDING)
+public class ClassOneTest {
+
+    @BeforeClass
+    public static void onceExecutedBeforeAll() {
+        System.out.println("@BeforeClass: executedBeforeAllTests");
```
Comment (tatsianabeliai) [inline src/test/java/ClassOneTest.java]: Fixed, added logger.

File: src/test/java/ClassOneTest.java (hunk 1 above)
Comment (tatsianabeliai) [inline src/test/java/ClassOneTest.java]: Fixed all unnecessary math variables

File: src/test/java/DDTTest.java (hunk 6 above)
Comment (tatsianabeliai) [inline src/test/java/DDTTest.java]: LOG is used in a static BeforeClass and AfterClass methods

File: src/test/java/DDTTest.java [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+    @Parameterized.Parameter(5)
+    public int expectedDiv;
+
+
+    @Parameterized.Parameters
+    public static Collection testData() throws IOException {
+        return getTestData("/testData.csv");
+    }
+
+    public static Collection<Object[]> getTestData(String filename) throws IOException {
```
Comment (tatsianabeliai) [inline src/test/java/DDTTest.java]: Fixed, added try-catch

File: src/test/java/ClassOneTest.java [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+        System.out.println("@After: executedAfterEachTest");
+    }
+
+    @AfterClass
+    public static void onceExecutedAfterAll() {
+        System.out.println("@AfterClass: executedAfterAllTests");
+    }
+
+
+    public final static int x = 50;
```
Comment (tatsianabeliai) [inline src/test/java/ClassOneTest.java]: Fixed constant declaration, added "telling" variable names

File: src/test/java/ClassOneTest.java [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+    public final static int y = 5;
+
+    @Category(SlowTests.class)
+    @Test
+    public void addTest() {
+        Mathematics mathematics = new Mathematics();
```
Comment (tatsianabeliai) [inline src/test/java/ClassOneTest.java]: done

File: src/test/java/ClassOneTest.java [hunk 12]
```diff
@@ -0,0 +1,81 @@
...
+        mathematics.add(x, y);
+        int result = mathematics.getResult();
+        assertEquals("Not Expected sum result", 55, result);
+
+    }
+
+    @Category(SlowTests.class)
+    @Test(timeout = 1000)
+    public void deductTest() throws InterruptedException {
```
Comment (tatsianabeliai) [inline src/test/java/ClassOneTest.java]: done

File: src/test/java/ClassOneTest.java [hunk 13]
```diff
@@ -0,0 +1,81 @@
...
+        Mathematics mathematics = new Mathematics();
+        mathematics.setResult(1);
```
Comment (tatsianabeliai) [inline src/test/java/ClassOneTest.java]: fixed

**Pull request merged**